### PR TITLE
Makes `PluralOperands::n` into a getter

### LINKED ISSUE #283: `PluralOperands::n` should become a getter
This is based on the Slack conversation.

tl;dr: we established that keeping a public field `n: f64` that is derived from what essentially are fixed point data represented with integers is a source of potential precision loss, which isn't justified, given that the field isn't used.

While we could do more to make the API even nicer, an easy change we can do right away is to change `n` to be a getter.

This is a backwards incompatible change, which we should be OK with as all potential uses are in the `icu4x` repository.

### PR DESCRIPTION
**Pull request by filmil**
The field `n` is not used for plural selection, and may cause
precision loss in cases of conversion.  So we stop keeping in
around and compute it on demand instead.

Fixes #283

### CODE REVIEW

File: components/pluralrules/src/rules/resolver.rs [hunk 1]
```diff
@@ -49,7 +49,7 @@ fn test_relation(relation: &ast::Relation, operands: &PluralOperands) -> bool {
 fn calculate_expression(expression: &ast::Expression, operands: &PluralOperands) -> Option<u64> {
     let value = match expression.operand {
         ast::Operand::N => {
-            if operands.n.fract() < f64::EPSILON {
+            if operands.n().fract() < f64::EPSILON {
```
Comment (sffc) [inline components/pluralrules/src/rules/resolver.rs:52]: Can you check `w == 0` here instead?

Comment (filmil) [reply components/pluralrules/src/rules/resolver.rs:52]: In that case, why have `n` to begin with?

Comment (sffc) [reply components/pluralrules/src/rules/resolver.rs]: Maybe we should just delete it altogether.  @zbraniecki?

Comment (zbraniecki) [reply components/pluralrules/src/rules/resolver.rs]: My rationale for keeping `n` is to:
* conform to UTS #35 
* allow someone to work with examples and test against them
* provide a convenient way to display the number that the operands are for. If we know that it is not used in runtime calculations (which it seems like!) then keeping it as a getter gives us a cheap way to have it.

It also is my understanding that DCE will make it cost nothing in use cases where noone calls it.

I'd prefer to keep it, but I'm open to be convinced to remove it in 0.2 as we learn about the uses

Comment (sffc) [reply components/pluralrules/src/rules/resolver.rs]: OK, let's keep it.

Dead code elimination should work as long as PluralOperands is a struct.  If it becomes a trait object, I think the compiler won't be able to eliminate it via DCE.  That's maybe one reason to keep this as a struct.

**Review by zbraniecki — APPROVED** [review]
lgtm! thank you!

**Comment by zbraniecki** [comment]
Good news!

```
     Running /Users/zbraniecki/projects/icu4x/target/release/deps/operands-602505516f95b50c
Gnuplot not found, using plotters backend
operands/create/usize   time:   [76.358 ns 76.982 ns 77.758 ns]
                        change: [-4.8303% -3.2631% -1.8373%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 1 outliers among 100 measurements (1.00%)
  1 (1.00%) high mild

operands/create/isize   time:   [81.892 ns 82.124 ns 82.382 ns]
                        change: [-11.014% -10.146% -8.9712%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 11 outliers among 100 measurements (11.00%)
  1 (1.00%) low mild
  4 (4.00%) high mild
  6 (6.00%) high severe

operands/create/string  time:   [670.61 ns 673.35 ns 676.65 ns]
                        change: [-4.9677% -4.1966% -3.3370%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 14 outliers among 100 measurements (14.00%)
  2 (2.00%) low mild
  5 (5.00%) high mild
  7 (7.00%) high severe
```

```
plurals/convert+select/fs
                        time:   [1.6578 us 1.6666 us 1.6767 us]
                        change: [-9.2869% -8.5142% -7.7610%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 8 outliers among 100 measurements (8.00%)
  5 (5.00%) high mild
  3 (3.00%) high severe

plurals/select/fs       time:   [1.6636 us 1.6709 us 1.6788 us]
                        change: [-8.0884% -7.3932% -6.6827%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 6 outliers among 100 measurements (6.00%)
  3 (3.00%) high mild
  3 (3.00%) high severe
```

Getting a 7-8% speed up (likely from the switch to `w`, but we noticed it because of it!) and shrinking the size of `PluralOperands` are both really nice wins! Thank you @sffc and @filmil for driving that change!

**Review by sffc — APPROVED** [review]

**Pull request merged**